src/types.ts

```diff
@@ -43,25 +43,47 @@ export type RemoteResourceWithoutFile = Optionalize<
   'filePrefix'
 >;
 
+/** Penumbra file composition */
+export type PenumbraFile = {
+  /** Backing stream */
+  stream: ReadableStream;
+  /** File mimetype */
+  mimetype: string;
+  /** Filename (excluding extension) */
+  filePrefix: string;
+  /** Relative file path (needed for zipping) */
+  path: string;
+};
+
 /**
- * The type that is emitted as progress continues
+ * Progress event details
  */
-export type ProgressEmit = CustomEvent<{
+export type ProgressDetails = {
+  /** The URL downloading from */
+  url: string;
+  /** Progress type */
+  type: 'decrypt' | 'zip';
   /** Percentage completed */
   percent: number;
   /** Total bytes read */
   totalBytesRead: number;
   /** Total number of bytes to read */
   contentLength: number;
-  /** The URL downloading from */
-  url: string;
-}>;
+};
 
-/** File returned by penumbra.get() */
-export type PenumbraFile = File | ReadableStream;
+/**
+ * The type that is emitted as progress continues
+ */
+export type ProgressEmit = CustomEvent<ProgressDetails>;
 
-/** Files returned by penumbra.get() */
-export type PenumbraFiles = File[] | ReadableStream;
+/**
+ * The type that is emitted when penumbra is ready
+ * to be used
+ */
+export type PenumbraReady = CustomEvent<{
+  /** Penumbra API object */
+  penumbra: PenumbraAPI;
+}>;
 
 /** Compression levels */
 export enum compression {
```

Comment:
Can we follow the style-guide convention here and name enums using `PascalCase` -> `Compression`

and the keys should also be pascal case -> `High`

Reply:
Ok, I'm going through `s/penumbra/Penumbra/`